### DISCUSSION TITLE
glibc version issue while using prisma on ubuntu 23

### DISCUSSION THREAD

**Discussion by Warisaremou** [prisma/prisma#23300] [Q&A]
### Question

Hi I got this error while using prisma in Nextjs on ubuntu :

Unhandled Runtime Error Error: Invalid prisma.user.findUnique() invocation:

Unable to require(/media/waris/DEV/My courses/nextjs-prisma/blog-course/node_modules/.prisma/client/libquery_engine-debian-openssl-3.0.x.so.node). Prisma has detected an incompatible version of the glibc C standard library installed in your system. This probably means your system may be too old to run Prisma. Please refer to the documentation about Prisma's system requirements: https://pris.ly/d/system-requirements

Details: /snap/core20/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /lib/x86_64-linux-gnu/libssl.so.3)

![Capture d’écran du 2024-02-27 21-53-51](https://github.com/prisma/prisma/assets/107433692/b3e52209-01f2-4d44-9031-6cbe064f443f)

My ubuntu version is 23.10. Can someone help me pls?
node version: 18.19.1
prisma version: "^5.9.1"

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id        String    @id @default(uuid())
  email     String    @unique
  password  String
  name      String?
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  Post      Post[]
  Comment   Comment[]
}

model Post {
  id        String    @id @default(uuid())
  image     String?
  title     String
  content   String
  likes     Int       @default(0)
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  user      User      @relation(fields: [userId], references: [id])
  userId    String
  Comment   Comment[]
}

model Comment {
  id        String   @id @default(uuid())
  content   String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  user      User     @relation(fields: [userId], references: [id])
  userId    String
  post      Post     @relation(fields: [postId], references: [id])
  postId    String
}
```

```ts
import { PrismaClient } from "@prisma/client";

const globalForPrisma = global as unknown as { prisma: PrismaClient };

export const prisma =
	globalForPrisma.prisma ||
	new PrismaClient({
		log: ["query"],
	});

if (process.env.NODE_ENV === "production") globalForPrisma.prisma = prisma; 

// Add any relevant Prisma Client queries here
```

- OS:  
- Database: 
- Node.js version: 
v18.19.1
- Run `prisma -v` to see your Prisma version and paste it
"prisma": "^5.9.1",

**Comment by ludralph** [comment]
Hi @Warisaremou 👋 

Thank you for raising this question.

Can you try specifying `binaryTargets` in the `generator` block of your `schema.prisma` file and setting it to `native`?
```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}
```

**Comment by Warisaremou** [comment]
@ludralph I add it but still not working
![Capture d’écran du 2024-02-28 08-30-30](https://github.com/prisma/prisma/assets/107433692/17683929-d349-4bd5-92ea-304bceb75f97)

**Reply by ludralph** [reply]
@Warisaremou, have you tried the suggestion in this [ask ubuntu](https://askubuntu.com/questions/1505918/glibc-version-issue-while-using-prisma-on-ubuntu) forum? I can see we have support for ubuntu up until [version 23.04](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#linux-ubuntu-x86_64) but you are using 23.10.

**Reply by Warisaremou** [reply]
Yes, someone answer me

![Capture d’écran du 2024-02-28 11-44-34](https://github.com/prisma/prisma/assets/107433692/69d87e41-c2f8-4e28-899c-3246452e397c)

I think it's because of my version of ubuntu

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
